pkg/aggregator/aggregator.go

```diff
@@ -865,7 +865,10 @@ func (agg *BufferedAggregator) tags(withVersion bool) []string {
 		}
 	}
 	if agg.haAgent.Enabled() {
-		tags = append(tags, "agent_group:"+agg.haAgent.GetGroup())
+		tags = append(tags,
+			"agent_group:"+agg.haAgent.GetGroup(),
+			"agent_role:"+string(agg.haAgent.GetRole()),
```

Comment:
Please don't add dynamic tags to the agent_running metric - distinct tags are considered to be distinct time series by the backend, and this will cause both series to be reported for a period of time around tag change (due to interpolation), causing confusion and erroneous aggregates.

To avoid querying issues, I would recommend creating a boolean gauge metric (ex. `datadog.agent.ha_agent.is_leader`), where the value, rather than the tags, indicate the agent state.